xblock/mixins.py

```diff
@@ -516,6 +516,10 @@ def _add_field(self, node, field_name, field):
         as an xml attribute or creates a separate child node.
         """
         value = field.to_string(field.read_from(self))
+
+        if not isinstance(value, basestring):
```

Comment:
@e-kolpakov The thing that bothers me is that with this approach is that it feels that if the `add_xml_to_node` takes care of serializing `None` values, I would also expect `parse_xml` to take care of converting the serialized values back to `None`. Right now taking care of `None` values is split between the general serialization method (`add_node_to_xml`) and individual fields which will have to make sure to convert empty strings back to `None` in their `from_json` implementation.

I understand your desire to keep the impact minimal though, so I'm fine with this if you think doing it inside `DateTime.to_json` could have unexpected side effects.